### PR TITLE
[receiver/mysql]: add more mysql metrics

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

This PR is combination of several already approved PRs. It contains only PRs with all conversations resolved:

  - Add mysql.connection.errors metric (#14723)
  - Add mysql.mysqlx_connections metric (#14727)
  - Add mysql.joins metric (#14728)
  - Add mysql.table_open_cache metric (#14737)
  - Add mysql queries (all, client and slow) count metrics (#14738)
  - Add metrics based on events_statements_summary_by_digest table (#14770)
  - Add mysql.client.network.io metric (#14744)

This is to speed up process and avoid rebase iterations

**Link to tracking Issue:**
#14138

**Testing:**
unit tests

**Documentation:**
`metadata.yaml`

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
@djaglowski @codeboten Please look at it

File: receiver/mysqlreceiver/metadata.yaml [hunk 1]
```diff
@@ -359,3 +406,40 @@ metrics:
       monotonic: false
       aggregation: cumulative
     attributes: [mysqlx_threads]
+  mysql.table_open_cache:
+    enabled: true
+    description: The number of statuses (e.g. hit) for open tables cache lookups.
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:411]: Wording is confusing to me. Can you rephrase?

File: receiver/mysqlreceiver/metadata.yaml [hunk 2]
```diff
@@ -359,3 +406,40 @@ metrics:
...
+    unit: 1
+    sum:
+      value_type: int
+      input_type: string
+      monotonic: true
+      aggregation: cumulative
+    attributes: [cache_status]
+  mysql.queries.client.count:
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:419]:
```suggestion
  mysql.query.client.count:
```

File: receiver/mysqlreceiver/metadata.yaml [hunk 3]
```diff
@@ -359,3 +406,40 @@ metrics:
...
+    enabled: true
+    description: The number of statements executed by the server. This includes only statements sent to the server by clients.
+    unit: 1
+    sum:
+      value_type: int
+      input_type: string
+      monotonic: true
+      aggregation: cumulative
+  mysql.queries.count:
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:428]:
```suggestion
  mysql.query.count:
```

File: receiver/mysqlreceiver/metadata.yaml [hunk 4]
```diff
@@ -359,3 +406,40 @@ metrics:
...
+    enabled: true
+    description: The number of statements executed by the server.
+    unit: 1
+    sum:
+      value_type: int
+      input_type: string
+      monotonic: true
+      aggregation: cumulative
+  mysql.queries.slow.count:
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:437]:
```suggestion
  mysql.query.slow.count:
```

File: receiver/mysqlreceiver/metadata.yaml [hunk 5]
```diff
@@ -320,6 +336,37 @@ metrics:
...
+    description: Errors that occur during the client connection process.
+    unit: 1
+    sum:
+      value_type: int
+      input_type: string
+      monotonic: true
+      aggregation: cumulative
+    attributes: [connection_error]
+  mysql.mysqlx_connections:
+    enabled: true
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:350]: We need to start making new metrics as disabled by default. The exception is if they are considered particularly valuable, then we have to use a featuregate (if the component is beta or stable).

File: receiver/mysqlreceiver/metadata.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply receiver/mysqlreceiver/metadata.yaml]: Is it better now?

File: receiver/mysqlreceiver/metadata.yaml (hunk 5 above)
Comment (sumo-drosiek) [reply receiver/mysqlreceiver/metadata.yaml:350]: Disabled all new metrics by default. We can enable them as defaults in separate PRs

**Review by djaglowski — APPROVED** [review]

**Comment by djaglowski** [comment]
Thanks @sumo-drosiek for all the contributions, patience, and for bundling approved PRs like this.
